File: Interview Preparation Kit/Search/IceCreamParlor.java
// https://www.hackerrank.com/challenges/ctci-ice-cream-parlor/problem
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the whatFlavors function below.
    static void whatFlavors(int[] cost, int money) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < cost.length; i++) {
            map.put(cost[i], i);
        }

        for (int j = 0; j < cost.length; j++) {
            int first_price = cost[j];
            int second_price = money - first_price;
            if (map.containsKey(second_price) && map.get(second_price) != j) {
                int first_index = j + 1;
                int second_index = map.get(second_price) + 1;
                System.out.println(first_index + " " + second_index);
                break;
            }
        }
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int t = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        for (int tItr = 0; tItr < t; tItr++) {
            int money = scanner.nextInt();
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

            int n = scanner.nextInt();
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

            int[] cost = new int[n];

            String[] costItems = scanner.nextLine().split(" ");
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

            for (int i = 0; i < n; i++) {
                int costItem = Integer.parseInt(costItems[i]);
                cost[i] = costItem;
            }

            whatFlavors(cost, money);
        }

        scanner.close();
    }
}

File: Interview Preparation Kit/Search/IceCreamParlor.js
// https://www.hackerrank.com/challenges/ctci-ice-cream-parlor/problem
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the whatFlavors function below.
function whatFlavors(cost, money) {
    let map = new Map();

    for (let i = 0; i < cost.length; i++) {
        map.set(cost[i], i);
    }

    for (let j = 0; j < cost.length; j++) {
        let first_price = cost[j];
        let second_price = money - first_price;
        if (map.has(second_price) && map.get(second_price) != j) {
            let first_index = j + 1;
            let second_index = map.get(second_price) + 1;
            console.log(first_index + " " + second_index);
            break;
        }
    }
}

function main() {
    const t = parseInt(readLine(), 10);

    for (let tItr = 0; tItr < t; tItr++) {
        const money = parseInt(readLine(), 10);

        const n = parseInt(readLine(), 10);

        const cost = readLine().split(' ').map(costTemp => parseInt(costTemp, 10));

        whatFlavors(cost, money);
    }
}
